connect/rest-extension/src/main/java/org/apache/kafka/connect/rest/extension/BasicAuthSecurityRestExtension.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.rest.extension;
+
+import org.apache.kafka.common.utils.AppInfoParser;
+import org.apache.kafka.connect.rest.ConnectRestExtension;
+import org.apache.kafka.connect.rest.ConnectRestExtensionContext;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * {@link BasicAuthSecurityRestExtension} provides the ability to authenticate incoming BasicAuth
+ * credentials using the configured JAAS {@link javax.security.auth.spi.LoginModule}.
+ * An entry with the name KafkaConnect is expected in the JAAS config file configured in the JVM.
```

Comment:
Use quotes or `{@code }` around "KafkaConnect" to highlight the importance of that literal.

Reply:
How about also mentioned that it must be configured in the worker configuration, and providing an example configuration fragment that shows how to use this sample extension.

Reply:
Should this mention that this is a sample implementation? How does it work, and where does it get the credentials? Seems like this needs a lot more context to be useful as an example.